squad-server/index.js

```diff
@@ -288,6 +288,10 @@ export default class SquadServer extends EventEmitter {
     this.logParser.on('TICK_RATE', (data) => {
       this.emit('TICK_RATE', data);
     });
+
+    this.logParser.on('SQUAD_CREATED', (data) => {
+      this.emit('SQUAD_CREATED', data);
```

Comment:
Seams like a simple add. 

Issue, we are getting the Squad ID in two places that may not match; player.squadID and data.squadID. 

I'm not sure of any race/issues that would crop up in the case of a player who Joins, and Immediately makes a squad before we update the player list. 

I figured just firing off the event with what we know is true, and let other people bind/request the information if they need more inside of their plugins. 

I'll work on this tomorrow/this evening.

Reply:
> Would it be worth it to update the player object here?
Fair point, I wonder whether the RCON response for `ListPlayers` will be update by this point. If so then we can force refresh it when using `getPlayerBySteamID`.

Reply:
The reason I think this is a potential issue has to do with Map Changes; the majority of squads are created right after a map change.

+ FuzzMT created SQ: 1-Russian Ground Forces
HELI
[12:51 AM]
+ Jumpwer created SQ: 1-United States Army
HELI
[12:51 AM]
+ Yokahigh created SQ: 2-Russian Ground Forces
BMP
[12:51 AM]
+ PhD Caboose created SQ: 2-United States Army
Squad 2
[12:51 AM]
+ Oak created SQ: 3-Russian Ground Forces
ARMOR
[12:51 AM]
+ RESI_STONER created SQ: 3-United States Army
HELI FOR ROLL OUT ONLY
[12:51 AM]
+ whoopieman created SQ: 4-United States Army
ARMOR NO CHILL
[12:52 AM]
+ Dren created SQ: 4-Russian Ground Forces
OTHER HELI
[12:52 AM]
+ Putin created SQ: 5-Russian Ground Forces
MIC REQ
[12:52 AM]
+ Bulltwinkle created SQ: 5-United States Army
ARMOR

Depends on time of day, but when our server has our "regular" crowd on, Squad are created fairly quickly as SL's load in. I feel like this is where the window between ListPlayers might be an issue, especially since map changes are also the volatile period for people leaving/joining.

PLAYER_CONNECTED doesn't update the player list, or force an update from what I can see. So in theory, a call to getPlayerBySteamID might not resolve. Forcing an update in this case is probably best.

Reply:
By force refresh, or rather force update, I meant:
```js
data.player = await this.getPlayerBySteamID(data.playerSteamID, true); // Not the "true" as the second parameter.
```

Rather than:
```js
data.player = await this.getPlayerBySteamID(data.playerSteamID);
data.player.squadID = data.squadID;
```

The former will force SquadJS to send the `ListPlayers` command to the server and then update the internal player list, effectively overriding the standard 30-second update interval. Whilst both are technically correct the latter is a break from the norm. I think there are also potential benefits to the former, for example, when a new squad is created it's likely to be populated quickly with players so by forcing the update using `ListPlayers` you will also update the stored `squadID` of those players too.

> PLAYER_CONNECTED doesn't update the player list, or force an update from what I can see. So in theory, a call to getPlayerBySteamID might not resolve. Forcing an update in this case is probably best.

Good spot! I think this being missed is the cause of some of the bug reports we've received about the `PLAYER_CONNECTED` event.

Reply:
Changed getPlayerBySteamID to force. 

Another question, is there any reason we would want to track a list of Team: SquadIDs : SquadNames? I think previously we only had the SquadID, not the actual Squad Name. 

I'm thinking about cases of Squad Baiting/Making a Squad and immediately leaving etc. We can't really track "Squad Leaders" persay, except when we get a "SQUAD_CREATED" event, we know they were Squad Leader at least for a short while. 

Its something that could be done in a plugin since I feel the information might not be the most reliable.

Reply:
> Another question, is there any reason we would want to track a list of Team: SquadIDs : SquadNames? I think previously we only had the SquadID, not the actual Squad Name.

Yeah, we discussed before keeping a list of squad info separately then embedding it into player objects too, but we never got around to implementing it.